ci/generate_json.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+readonly repo_root=$(dirname "$0")/..
+cd "$repo_root"
+
+PYTHONPATH="$repo_root/src" python3 -m rapids_metadata.json --all-versions > rapids-metadata.json
```

Comment:
```suggestion
PYTHONPATH="$repo_root/src" python3 -m rapids_metadata.json --all-versions | jq . > rapids-metadata.json
```

Would you consider pretty-printing this output? That'd have the nice side effect of making diffs in PRs a bit easier to read. And I don't think the extra data size from the added whitespace would be prohibitively large.

Reply:
I don't know if `pre-commit` has a way to install `jq` as a dependency. Python's `json` library does have pretty-print functionality, but do we want to always enable that? Or maybe with a `--pretty` parameter?

Reply:
This is a local hook so we could just install `jq` in CI + assume everyone developing on the project has a system installation of it.

But I don't feel that strongly about `jq` being the specific mechanism. A `--pretty` parameter sounds good to me (I don't think we'd want it to be the default).

Reply:
Try this: https://github.com/pre-commit/pre-commit-hooks?tab=readme-ov-file#pretty-format-json

Reply:
❗ I like that even better, let's do that instead of adding a parameter.